### DISCUSSION TITLE
Suspense And Loading

### DISCUSSION THREAD

**Discussion by soblvsk** [vercel/next.js#58865] [App Router] [answered]
Hello everyone, I decided to create such a discussion, since I myself encountered this today in next 14.

Faced with such a situation:
I was making a catalog for an online store and wanted to have skeletons for goods while I was waiting for all the goods to be received from the backup. But Suspense didn't work at all.

I had to create a loading.tsx file and copy all the code from page.tsx and where I had a component with goods, replace it with Skeleton.

And I would like to know if this is normal or if there is some explanation for this. Do you have any options?

At the bottom is a part of the code that I eventually had to do, but I only wanted to use Suspense without loading.tsx file

_**page.tsx**_
```js
export default async function CatalogPage({ searchParams }: CatalogPageProps) {
// The rest of the code
  return (
      <section className="relative container my-10 flex flex-col space-y-6">
         <Filters query={query} sort={sort} />
        {/* <Suspense fallback={<CatalogSkeleton/>}> */}
         <Products products={filteredProducts} />
        {/* </Suspense> */}
          {filteredProducts.length ? (
            <Pagination
              totalPages={totalPages}
              currentPage={currentPage}
              sort={sort}
            />
          ) : null}
      </section>
  );
}
```

_**loading.tsx**_

```js
export default function CatalogLoading() {
  return (
    <>
      <section className="relative container my-10 flex flex-col space-y-6">
        <FiltersSkeleton />
        <ProductsSkeleton />
        <PaginationSkeleton />
      </section>
    </>
  );
}
```

**Comment by icyJoseph** [comment]
Do you do any async work within `Products`?

**Reply by soblvsk** [reply]
There is no async at all in the `Products` component. In `CatalogPage` itself I use `fetchProducts()`

**Reply by icyJoseph** [reply, marked as answer]
Yeah, but if you place the suspense boundary around Products, and Products is not doing any asynchronous work, then it never "suspends", so the fallback is never needed.

From the docs, https://react.dev/reference/react/Suspense:

> children: The actual UI you intend to render. If children suspends while rendering, the Suspense boundary will switch to rendering fallback.

In your case,  passing `filteredProducts` to `Products` doesn't suspend. It is a regular render pass. However, if you were to wrap the entire `CatalogPage` with a suspense boundary, then when its parent renders it, it'd find async work, it'd suspend, and use the fallback, or in your case, the `loading.tsx` file.
